### ISSUE TITLE
Needless redefined-variable-type: difference between instancemethod and function is uninteresting

### ISSUE THREAD

**Issue by nedbat** [PyCQA/pylint#772]
This code:

```
        if env.PY2:
            writeout = lambda u: outfile.write(u.encode(output_encoding()))
        else:
            writeout = outfile.write
```

produces:

```
coverage/summary.py:57: Redefinition of writeout type from function to instancemethod (redefined-variable-type)
```

The distinction between a function and an instancemethod is pedantic and uninteresting.  This shouldn't cause a warning.

**Comment by The-Compiler** [comment]
I'd expect this to be covered by the fix for #710.

**Comment by PCManticore** [comment]
Yes, this shouldn't be emitted. I'll check Laura's PR for integration in 1.5.4.

**Comment by nedbat** [comment]
BTW, I don't think the "if/else" aspect of this is important.  My point is that "instancemethod" and "function" are not interesting to distinguish.  This code shouldn't raise a warning either:
```
writeout = lambda u: outfile.write(u.encode(output_encoding()))
writeout = outfile.write
```

**Comment by PCManticore** [comment]
Closing this. redefined-variable-type is now an extension in pylint 1.7, which means it is disabled by default and has to be explicitly enabled. We will probably not spend more effort improving it.
